specviz/third_party/glue/viewer_single_spectrum.py

```diff
@@ -0,0 +1,170 @@
+import os
+
+import uuid
+import numpy as np
+from qtpy.QtWidgets import QWidget, QVBoxLayout, QCheckBox
+
+from glue.config import qt_client
+from glue.core.data_combo_helper import ComponentIDComboHelper
+
+from glue.external.echo import CallbackProperty, SelectionCallbackProperty
+from glue.external.echo.qt import (connect_checkable_button,
+                                   autoconnect_callbacks_to_qt)
+
+from glue.viewers.common.layer_artist import LayerArtist
+from glue.viewers.common.state import ViewerState, LayerState
+from glue.viewers.common.qt.data_viewer import DataViewer
+
+from glue.utils.qt import load_ui
+
+from .utils import glue_data_to_spectrum1d, is_glue_data_1d_spectrum
+from ...widgets.main_window import MainWindow
+
+__all__ = ['SpecvizSingleDataViewer']
+
+
+class SpecvizSingleViewerState(ViewerState):
+
+    y_att = SelectionCallbackProperty(docstring='The attribute to use on the y-axis')
+
+    def __init__(self, *args, **kwargs):
+        super(SpecvizSingleViewerState, self).__init__(*args, **kwargs)
+        self._y_att_helper = ComponentIDComboHelper(self, 'y_att')
+        self.add_callback('layers', self._on_layers_change)
+
+    def _on_layers_change(self, value):
+        self._y_att_helper.set_multiple_data(self.layers_data)
+
+
+class SpecvizSingleLayerState(LayerState):
+    pass
+
+
+class SpecvizSingleLayerArtist(LayerArtist):
+
+    _layer_state_cls = SpecvizSingleLayerState
+
+    def __init__(self, specviz_window, *args, **kwargs):
+
+        super(SpecvizSingleLayerArtist, self).__init__(*args, **kwargs)
+
+        self.specviz_window = specviz_window
+        self.plot_widget = self.specviz_window.workspace.current_plot_window.plot_widget
+
+        self.state.add_callback('visible', self.update)
+        self.state.add_callback('zorder', self.update)
+        self._viewer_state.add_callback('y_att', self.update)
+        self.uuid = str(uuid.uuid4())
+
+    def _on_visible_change(self, value=None):
+        self.redraw()
+
+    def _on_zorder_change(self, value=None):
+        self.redraw()
+
+    def clear(self):
+
+        # FIXME: simplify the following by implementing DataListModel.remove_data
+        # and making it able to take e.g. data name.
+
+        data_model = self.specviz_window.workspace._model
+        for i in range(data_model.rowCount()):
+            item = data_model.item(i)
+            if item.data() == self.uuid:
+                data_model.removeRow(i)
+                return
```

Comment:
@nmearl - this is where it would be handy to be able to remove a data object from Specviz based on its name (which here is a a UUID)

Reply:
I've implemented the method `data_model.remove_data(<id>)` in #441.